code/datums/components/climbable.dm

```diff
@@ -0,0 +1,101 @@
+///This components makes it possible to climb (forcemove yourself) onto an atom by clicking dragging onto it.
+/datum/component/climbable
+	///How long the climb act takes.
+	var/climb_time
+	///How long the climb act stuns the climber
+	var/climb_stun
+	///If the parent exhibits tablelike behaviour. Basically this means that by dragging your active item onto it, you place it onto the table.
+	var/table_behaviour
+	///What mobs are currently climbing the parent.
+	var/list/structureclimbers
+
+/datum/component/climbable/Initialize(climb_time = 20, climb_stun = 20, table_behaviour = TRUE)
+	if(!isatom(parent))
+		return COMPONENT_INCOMPATIBLE
+	structureclimbers = list()
+	src.climb_time = climb_time
+	src.climb_stun = climb_stun
+	src.table_behaviour = table_behaviour
+
+/datum/component/climbable/Destroy()
+	structureclimbers = null
+	return ..()
+
+/datum/component/climbable/RegisterWithParent()
+	..()
+	RegisterSignal(parent, list(COMSIG_ATOM_FORCED_MOVEMENT_BUMPED), .proc/on_forced_movement_bump)
+	RegisterSignal(parent, list(COMSIG_MOUSEDROPPED_ONTO), .proc/on_MouseDrop_T)
+	RegisterSignal(parent, list(COMSIG_ATOM_ATTACK_HAND), .proc/on_attack_hand)
+	if(istype(parent, /obj/structure/closet))
+		RegisterSignal(parent, list(COMSIG_CLOSET_CLOSE), .proc/on_closet_close)
+		RegisterSignal(parent, list(COMSIG_CLOSET_OPEN), .proc/on_closet_open)
+
+///Handles the actual act of moving
+/datum/component/climbable/proc/do_climb(atom/source, atom/movable/A)
+	A.forceMove(get_step_towards(A, source))
```

Comment:
You need to check somewhere that there isn't something else preventing the movement

Reply:
Reimplemented the old hacky thing because I couldn't think of anything better